### PR TITLE
Fixed bug in rest.py 

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

File: raiden/api/rest.py [hunk 1]
```diff
@@ -562,6 +562,15 @@ def open(
             settle_timeout=settle_timeout,
         )
         try:
+            token = self.raiden_api.raiden.chain.token(token_address)
```
Comment (LefterisJP) [inline raiden/api/rest.py]: This part does not need to be inside the `try:`:

File: raiden/api/rest.py [hunk 2]
```diff
@@ -562,6 +562,15 @@ def open(
...
+            balance = token.balance_of(self.raiden_api.raiden.address)
+            if total_deposit > balance:
```
Comment (LefterisJP) [inline raiden/api/rest.py]: I suggest to change the code you added from what you have in the PR to:

```python
    if total_deposit is not None and total_deposit > balance:
        error_msg =  'Not enough balance to deposit. {} Available={} Needed={}'.format(
            pex(token_address),
            balance,
            total_deposit,
        )
        return api_error(
            errors=error_msg,
            status_code=HTTPStatus.PAYMENT_REQUIRED,
        )
```

1. The reason is that `total_deposit` can also be `None` in which case you don't need to check anything.
2. To return an error from the API you need to use `return api_error()`

**Comment by LefterisJP** [comment]
Also please amend your commit and remove the 3-4 old files you are trying to add back. Some images and an .rst file.

**Comment by Giulio2002** [comment]
i did what you asked, you can check my fork

**Comment by Giulio2002** [comment]
i applied the changes you asked by hand because github don't like your changes

File: raiden/api/rest.py [hunk 3]
```diff
@@ -561,7 +561,19 @@ def open(
             token_address=to_checksum_address(token_address),
             settle_timeout=settle_timeout,
         )
+        token = self.raiden_api.raiden.chain.token(token_address)
+        balance = token.balance_of(self.raiden_api.raiden.address)
         try:
+            if total_deposit is not None and total_deposit > balance:
```
Comment (LefterisJP) [inline raiden/api/rest.py]: as I said before please remove all of these additions from inside the `try:` clause. Move them back up where `balance` is.

File: raiden/api/rest.py [hunk 4]
```diff
@@ -561,7 +561,19 @@ def open(
...
+                error_msg =  'Not enough balance to deposit. {} Available={} Needed={}'.format(
```
Comment (LefterisJP) [inline raiden/api/rest.py]: Here there is an extra space after `=`. I recommend installing `flake8` and running it to make sure that you got no style violations.  For that and more please read the [contributing](https://github.com/raiden-network/raiden/blob/master/CONTRIBUTING.md) guide.

**Comment by LefterisJP** [comment]
Also please make sure that there is only 1 commit for this PR. Squash all the commits you are making.

And add a changelog entry too since this is a user facing bug. The changelog is [here](https://github.com/raiden-network/raiden/blob/dc9406351589f5b4ec2192db3a7a16c4ccf93008/docs/changelog.rst).
Add an entry in the beginning of the flile there: 
```
* :bug'`3022` Reject REST API channel opening with an error if there is not enough token balance for the initial deposit.
```

File: docs/changelog.rst [hunk 5]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (LefterisJP) [inline docs/changelog.rst]: Remove the empty line here.

File: docs/changelog.rst [hunk 6]
```diff
@@ -203,6 +204,7 @@ Changelog
 * :bug:`870` User selectable NAT traversal.
 * :feature:`921` Add ``/api/1/connection`` API endpoint returning information about all connected token networks.
 * :bug:`1011` Remove ``settled`` attribute from the NettingChannel smart contract.
+* :bug'`3022` Reject REST API channel opening with an error if there is not enough token balance for the initial deposit.
```
Comment (LefterisJP) [inline docs/changelog.rst]: Don't put this here but in the top of the file -- iow first entry after :release:`0.16.0 <2018-11-09>`

File: raiden/api/rest.py [hunk 7]
```diff
@@ -1174,3 +1188,4 @@ def patch_channel(
                 status_code=HTTPStatus.BAD_REQUEST,
             )
         return result
+
```
Comment (LefterisJP) [inline raiden/api/rest.py]: don't add arbitrary new line here.

File: raiden/api/rest.py [hunk 8]
```diff
@@ -561,6 +561,20 @@ def open(
             token_address=to_checksum_address(token_address),
             settle_timeout=settle_timeout,
         )
+        token = self.raiden_api.raiden.chain.token(token_address)
+        balance = token.balance_of(self.raiden_api.raiden.address)
+
+        if total_deposit is not None and total_deposit > balance:
+           error_msg = 'Not enough balance to deposit. {} Available={} Needed={}'.format(
```
Comment (LefterisJP) [inline raiden/api/rest.py]: fix indentation level here to be 4 spaces. As I said before you can check that this is okay before committing by running flake8 on your code.

File: raiden/api/rest.py [hunk 9]
```diff
@@ -561,6 +561,20 @@ def open(
...
+               pex(token_address),
+               balance,
+               total_deposit,
+           )
+           return api_error(
```
Comment (LefterisJP) [inline raiden/api/rest.py]: Same here with the indentation

**Comment by Giulio2002** [comment]
i applied your changes like your requested

**Comment by LefterisJP** [comment]
It's almost done. But I asked for the changelog entry to be **above** the 0.16.0 line.

**Comment by LefterisJP** [comment]
You also need to rebase on latest master @Giulio2002

**Comment by LefterisJP** [comment]
@Giulio2002 I said **rebase** on latest master -- not merge master onto your branch. We have a rebase only policy.

**Comment by LefterisJP** [comment]
@Giulio2002 If it's fine with you I can take over and just correct the branch a bit -- let the tests run and then merge it?

**Comment by Giulio2002** [comment]
okay, now i changed changelog.rst as you asked. and i rebased to the latest master as well. btw you can make any changes you want on my pr.

**Pull request merged**